src/community/jdbcstore/src/main/java/org/geoserver/jdbcstore/internal/JDBCDirectoryStructure.java

```diff
@@ -0,0 +1,406 @@
+/* Copyright (c) 2015 OpenPlans - www.openplans.org. All rights reserved.
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.jdbcstore.internal;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.sql.DataSource;
+
+import org.geoserver.platform.resource.Paths;
+import org.geoserver.jdbcstore.internal.JDBCQueryHelper.*;
+
+import static org.geoserver.jdbcstore.internal.JDBCQueryHelper.*; 
+
+/**
+ * 
+ * Handles database access & ORM mapping of directory structure
+ *
+ * @author Kevin Smith, Boundless
+ * @author Niels Charlier
+ *
+ *
+ */
+public class JDBCDirectoryStructure {
+    
+    private static final Logger LOGGER = org.geotools.util.logging.Logging.getLogger(JDBCDirectoryStructure.class);
+            
+    protected final static String TABLE_RESOURCE = "resource";
```

Comment:
RESOURCE is an Oracle reserved word. If you use this as a table name, you will not be able to use jdbcstore on Oracle without a great deal of quoting. I am not a DBA, but many will recommend using plurals for table names. How about changing this to ```TABLE_RESOURCES = "resources"```?